litgpt/finetune/adapter.py

```diff
@@ -226,7 +226,7 @@ def fit(
             t1 = time.perf_counter() - t0
             fabric.print(f"iter {iter_num}: val loss {val_loss.item():.4f}, val time: {t1 * 1000:.2f} ms")
             fabric.barrier()
-        if not is_accumulating and step_count % train.save_interval == 0:
+        if train.save_interval is not None and not is_accumulating and state["step_count"] % train.save_interval == 0:
```

Comment:
The state["step_count"] change seems to be an accident. Not all the files have a state

Reply:
Good catch, this should be fixed now. Tested that all these scripts run ok.